src/layers.cpp

```diff
@@ -793,50 +1189,55 @@ void UpdateLayers()
 
                 if(g_compatibility.enable_climb_bgo_layer_move)
                 {
-                    int allBGOs = numBackground + numLocked;
-                    for(B = 1; B <= allBGOs; B++)
+                    for(int B : Layer[A].BGOs)
                     {
-                        if(BackgroundFence[Background[B].Type] && Background[B].Layer == Layer[A].Name)
+                        if(BackgroundFence[Background[B].Type])
                         {
+                            // @Wohlstand, should this be 0?
```

Comment:
No, this is required for player to inherit the move speed to move together with fences

Reply:
But this is the section where the layer is stopped (see above `Block[B].SpeedX = 0`)

Reply:
I guess, then, should be right, but please, test these demos to make sure nothing got broken:
[fence-tests.zip](https://github.com/Wohlstand/TheXTech/files/7972644/fence-tests.zip)
You may toggle between modern and vanilla modes to see the behavior differences

Reply:
I will think more about these later -- we may also need to add extra test cases (starting and stopping the layer's motion, attaching the layer to an NPC, etc) to test all of the unusual cases I'm discussing here. Let's not deal with that during this merge.